Sources/NIOCore/ByteBuffer-aux.swift

```diff
@@ -156,6 +215,19 @@ extension ByteBuffer {
         self._moveReaderIndex(forwardBy: length)
         return result
     }
+    
+    /// Read a null terminated string off this `ByteBuffer`, decoding it as `String` using the UTF-8 encoding. Move the reader index
+    /// forward by the string's length and its null terminator.
+    ///
+    /// - returns: A `String` value deserialized from this `ByteBuffer` or `nil` if there isn't a null terminated string readable.
```

Comment:
This could be clearer: it would be better to say that this returns `nil` if there is not a complete null-terminated string, including null-terminator, in the readable bytes of the buffer.